source/agora/common/Transaction.d

```diff
@@ -306,66 +307,113 @@ public Transaction newCoinbaseTX (PublicKey address, Amount value = 0)
 
 /*******************************************************************************
 
-    Create a set of chained transactions, where each transaction spends the
-    entire sum of the previous transaction.
+    Create a set of transactions, where each newly created transaction
+    spends the entire sum of each provided transaction's output as
+    set in the parameters.
 
-    Useful for quickly generating transactions for use with unittests.
+    If txes is null, the genesis transaction's outputs will be spent.
 
     Params:
-        root = the first transaction in the chain will spend the output of
-               this transaction
+        prev_txs = the previous transactions to refer to
         count = the number of transactions to create
         key_pair = the key pair used to sign transactions and to send
                    the output to
 
 *******************************************************************************/
 
 version (unittest)
-public Transaction[] getChainedTransactions (Transaction root, size_t count, KeyPair key_pair, size_t period = 1)
+public Transaction[] getChainedTransactions (KeyPair key_pair, size_t count,
```

Comment:
Since `count` *always* needs to be a multiple of `Block.TxsInBlock`, why not simplify the code and multiply `code` by `Block.TxsInBlock` ?
Also I don't like the `get` verb here. It's not "getting" anything, it is "make
ing, "build"ing, "create"ing, etc...

Reply:
Yes, so maybe `makeBlockTransactions`, and the `count` would specify the number of blocks?

Reply:
Ah actually it still returns transactions, so I'll just name it `makeTransactions` and document it better.

Reply:
I had something like `makeTransactionsToFillBlock` in mind, only it's too long.